kernel/libs/jhash/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+            slice[index + 5],
+            slice[index + 6],
+            slice[index + 7],
+        ]));
+        c = c.wrapping_add(u32::from_le_bytes([
+            slice[index + 9],
+            slice[index + 10],
+            slice[index + 11],
+            slice[index + 12],
+        ]));
+        index += 12;
+        length -= 12;
+    }
+
+    // Handle the last 12 bytes
+    if length == 12 {
+        c = c.wrapping_add((slice[index + 11] as u32) << 24);
+    }
+
+    if length >= 11 {
+        c = c.wrapping_add((slice[index + 10] as u32) << 16);
+    }
+
+    if length >= 10 {
+        c = c.wrapping_add((slice[index + 9] as u32) << 8);
+    }
+
+    if length >= 9 {
+        c = c.wrapping_add(slice[index + 8] as u32);
+    }
+
+    if length >= 8 {
+        b = b.wrapping_add((slice[index + 7] as u32) << 24);
+    }
+
+    if length >= 7 {
+        b = b.wrapping_add((slice[index + 6] as u32) << 16);
+    }
+
+    if length >= 6 {
+        b = b.wrapping_add((slice[index + 5] as u32) << 8);
+    }
+
+    if length >= 5 {
+        b = b.wrapping_add(slice[index + 4] as u32);
+    }
+
+    if length >= 4 {
+        a = a.wrapping_add((slice[index + 3] as u32) << 24);
+    }
+
+    if length >= 3 {
+        a = a.wrapping_add((slice[index + 2] as u32) << 16);
+    }
+
+    if length >= 2 {
+        a = a.wrapping_add((slice[index + 1] as u32) << 8);
+    }
+
+    if length >= 1 {
+        a = a.wrapping_add(slice[index] as u32);
+        return jhash_final(a, b, c);
+    }
+
+    c
+}
+
+/// Hashes exactly three u32 values
+pub const fn jhash_3vals(a: u32, b: u32, c: u32, initval: u32) -> u32 {
+    jhash_3vals_inner(
+        a,
+        b,
+        c,
+        initval.wrapping_add(JHASH_INITVAL).wrapping_add(3 << 2),
+    )
+}
+
+/// Hashes exactly two u32 values
+pub const fn jhash_2vals(a: u32, b: u32, initval: u32) -> u32 {
+    jhash_3vals_inner(
+        a,
+        b,
+        0,
+        initval.wrapping_add(JHASH_INITVAL).wrapping_add(2 << 2),
+    )
+}
+
+/// Hashes exactly one u32 value
+pub const fn jhash_1vals(a: u32, initval: u32) -> u32 {
+    jhash_3vals_inner(
+        a,
+        0,
+        0,
+        initval.wrapping_add(JHASH_INITVAL).wrapping_add(1 << 2),
+    )
+}
+
+/// Hashes an array of u32 values.
+pub const fn jhash_u32_array(array: &[u32], initval: u32) -> u32 {
+    let mut length = array.len() as u32;
+    let mut index = 0;
+
+    // Initialize values a, b, and c
+    let mut a: u32 = JHASH_INITVAL
+        .wrapping_add(length << 2)
+        .wrapping_add(initval);
+    let mut b: u32 = a;
+    let mut c: u32 = a;
+
+    // Process most values except the last three
+    while length > 3 {
+        a = a.wrapping_add(array[index]);
+        b = b.wrapping_add(array[index + 1]);
+        c = c.wrapping_add(array[index + 2]);
+
+        (a, b, c) = jhash_mix(a, b, c);
+        length -= 3;
+        index += 3;
+    }
+
+    // Handle the last three values
+    debug_assert!(length <= 3);
```

Comment:
Remove this `debug_assert`? The condition in the `while` statement clearly indicates `length <= 3` at this point.

(Otherwise, you might want to add a `debug_assert!(length <= 12)` in a similar position to `jhash_slice` for consistency. IMO, I don't think it's necessary.)